### PR TITLE
#5215 fix uncompleted coverage files generated

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5215)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
Only change config files
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/build.gradle [hunk 1]
```diff
@@ -115,7 +115,7 @@ task it {
         if(!project.hasProperty('ohara.manager.it.configurator')) throw new IllegalArgumentException("ohara.manager.it.configurator is required")
         String configurator = project.getProperty('ohara.manager.it.configurator')
 
-        commandLine toOsCommand('yarn', 'test:it:ci', '--port', 0, '--configurator', configurator)
+        commandLine toOsCommand('yarn', 'test:it:ci', '--configurator', configurator)
```
Comment (saivirtue) [inline ohara-manager/build.gradle:118]: We need to explicitly assign a server port for IT tests

File: ohara-manager/client/package.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
   },
   "scripts": {
     "setup": "yarn install --frozen-lockfile && yarn build",
-    "start": "react-scripts -r @cypress/instrument-cra start",
+    "start": "react-scripts start",
+    "start:ci": "react-scripts -r @cypress/instrument-cra start",
```
Comment (saivirtue) [inline ohara-manager/client/package.json:11]: separate client server commands without instrument code

File: ohara-manager/package.json [hunk 3]
```diff
@@ -29,16 +29,16 @@
     "test:coverage:check": "yarn --cwd client test:coverage:check",
     "test:e2e:open": "yarn --cwd client test:e2e:open",
     "test:e2e:run": "node ./scripts/runTests.js --testMode e2e",
-    "test:e2e:ci": "node ./scripts/runTests.js --testMode e2e --prod",
+    "test:e2e:ci": "node ./scripts/runTests.js --testMode e2e",
     "test:it:open": "yarn --cwd client test:it:open",
     "test:it:run": "node ./scripts/runTests.js --testMode it",
-    "test:it:ci": "node ./scripts/runTests.js --testMode it --prod",
+    "test:it:ci": "node ./scripts/runTests.js --testMode it --ci",
```
Comment (saivirtue) [inline ohara-manager/package.json:35]: We only need to run client server in same command for `test:it:ci` now

File: ohara-manager/package.json [hunk 4]
```diff
@@ -29,16 +29,16 @@
...
     "test:api:open": "yarn --cwd client test:api:open",
     "test:api:run": "yarn --cwd client test:api:run",
     "test:api:ci": "node ./scripts/runTests.js --testMode api",
     "clean": "run-s clean:processes clean:files",
     "clean:processes": "node ./scripts/cleanAllProcesses.js",
-    "clean:files": "rm -rf ./test-reports && rm -rf ./client/build && rm -rf ./client/coverage",
+    "clean:files": "rm -rf ./test-reports && rm -rf ./client/build && rm -rf ./client/coverage && rm -rf ./client/.nyc_output",
```
Comment (saivirtue) [inline ohara-manager/package.json:41]: The generated temp coverage files (nyc_output) need to be removed as weill

File: ohara-manager/build.gradle (hunk 1 above)
Comment (konekoya) [reply ohara-manager/build.gradle:118]: Because it's a fake configurator 🤔

File: ohara-manager/scripts/runTests.js [hunk 5]
```diff
@@ -79,18 +79,16 @@ const run = async (prod, apiRoot, serverPort = 5050, clientPort = 3000) => {
   // Wait until the server is ready
   await utils.waitOnService(`http://localhost:${serverPort}`);
 
-  // Start client server for e2e or it tests
-  // this server only starts on local env not on jenkins
-  if (!prod && testMode !== 'api') {
+  // Start client server for generating instrument code
+  // by instrument-cra plugin
+  // This instrument plugin is only available on development environment!
```
Comment (konekoya) [inline ohara-manager/scripts/runTests.js:84]: _in_ :)

File: ohara-manager/scripts/runTests.js [hunk 6]
```diff
@@ -79,18 +79,16 @@ const run = async (prod, apiRoot, serverPort = 5050, clientPort = 3000) => {
...
-      'forever',
-      ['start', 'node_modules/react-scripts/scripts/start.js'],
-      {
-        cwd: 'client',
-        stdio: 'inherit',
-      },
-    );
+    client = execa('yarn', ['start:ci'], {
+      cwd: 'client',
+      stdio: 'inherit',
```
Comment (konekoya) [inline ohara-manager/scripts/runTests.js:90]: @saivirtue So the process can be properly terminated after each test run, right?

Comment (konekoya) [reply ohara-manager/scripts/runTests.js:90]: I will file an issue to completely remove forever "forever" from our project XD

Comment (saivirtue) [reply ohara-manager/scripts/runTests.js:90]: In my laptop, only `forever` processes could not be terminated normally. So yes :)

Comment (konekoya) [reply ohara-manager/scripts/runTests.js:90]: Nice, thanks :) #5246

File: ohara-manager/build.gradle (hunk 1 above)
Comment (saivirtue) [reply ohara-manager/build.gradle:118]: assign a "zero" port will generate a random port for us, which could not be used for 5050 server if we ran client server in the same time :)

**Comment by saivirtue** [comment]
retry manager-it

File: ohara-manager/scripts/runTests.js [hunk 7]
```diff
@@ -79,16 +79,26 @@ const run = async (prod, apiRoot, serverPort = 5050, clientPort = 3000) => {
...
+  if (ci && testMode !== 'api') {
     console.log(chalk.blue(`Starting client server`));
     client = execa(
       'forever',
-      ['start', 'node_modules/react-scripts/scripts/start.js'],
+      [
+        'start',
+        '-c',
+        'node -r @cypress/instrument-cra',
+        'node_modules/react-scripts/scripts/start.js',
```
Comment (saivirtue) [inline ohara-manager/scripts/runTests.js:94]: This part drives me crazy... 😵 
we need to pass instrument plugin as argument of `node`...

**Comment by saivirtue** [comment]
retry manager-it

**Comment by saivirtue** [comment]
retry manager-e2e

**Pull request merged**